### PR TITLE
ogr_time

### PR DESCRIPTION
**Pull request by michaelmattig**

### CODE REVIEW

**Comment by michaelmattig** [comment]
[Daten jeweils mit OGR-Quelle laden](https://trello.com/c/HKcsvtu3/128-daten-jeweils-mit-ogr-quelle-laden)

**Review by ChristianBeilschmidt — CHANGES_REQUESTED** [review]

File: operators/src/source/ogr_source.rs [hunk 1]
```diff
@@ -145,12 +145,12 @@ pub enum OgrSourceTimeFormat {
         custom_format: String,
     },
     Seconds,
-    Iso,
+    Auto,
```
[the remark this answers is not in the document]
Comment (michaelmattig) [reply operators/src/source/ogr_source.rs:148]: `Auto` is for everything Gdal can parse which also includes Iso but also many other formats. I see no need to parse Iso ourselfs. So my idea was to use Auto if Gdal can parse it and custom if we parse it ourselfs.

File: operators/src/source/ogr_source.rs [hunk 2]
```diff
@@ -497,18 +497,27 @@ where
...
     fn create_time_parser(
         time_format: &OgrSourceTimeFormat,
-    ) -> Box<dyn Fn(&str) -> Result<TimeInstance> + '_> {
+    ) -> Box<dyn Fn(FieldValue) -> Result<TimeInstance> + '_> {
         match time_format {
-            OgrSourceTimeFormat::Iso => Box::new(move |date: &str| {
-                let date_time = DateTime::parse_from_rfc3339(date)?;
-                Ok(date_time.timestamp_millis().try_into()?)
+            OgrSourceTimeFormat::Auto => Box::new(move |field: FieldValue| {
+                // TODO: support date without time as well?
```
[the remark this answers is not in the document]
Comment (michaelmattig) [reply operators/src/source/ogr_source.rs]: 17bf927e34a21377ecf24eabffecf2158f7a44ee

File: services/Cargo.toml [hunk 3]
```diff
@@ -19,7 +19,7 @@ bytes = "1.0"
 chrono = { version = "0.4", features = ["serde"] }
 config = "0.11"
 futures = "0.3"
-gdal = { version = "0.7" }
+gdal = { version = "0.7", features = ["datetime"] }
```
[the remark this answers is not in the document]
Comment (michaelmattig) [reply services/Cargo.toml:22]: we are on it, but the CI makes problems. No idea what to do there: https://github.com/geo-engine/geoengine/pull/215

File: operators/src/source/ogr_source.rs [hunk 4]
```diff
@@ -497,18 +497,27 @@ where
...
+                    .into_string()
+                    .ok_or(Error::TimeIntervalColumnNameMissing)?;
+                let date_time = DateTime::parse_from_str(&date, &custom_format)?;
                 Ok(date_time.timestamp_millis().try_into()?)
             }),
-            OgrSourceTimeFormat::Seconds => Box::new(move |date: &str| {
-                let date_time = DateTime::parse_from_str(date, "%C")?;
+            OgrSourceTimeFormat::Seconds => Box::new(move |field: FieldValue| {
+                let date = field
+                    .into_string()
```
[the remark this answers is not in the document]
Comment (michaelmattig) [reply operators/src/source/ogr_source.rs]: 17bf927e34a21377ecf24eabffecf2158f7a44ee

File: operators/test-data/vector/data/points_with_iso_time.json [hunk 5]
```diff
@@ -58,8 +58,8 @@
         "coordinates": [19.171284, 50.636947]
       },
       "properties": {
-        "time_start": "+2014-05-01T00:00:00Z",
-        "time_end": "+2014-06-01T00:00:00Z"
+        "time_start": "2014-05-01T00:00:00Z",
+        "time_end": "2014-06-01T00:00:00Z"
```
[the remark this answers is not in the document]
Comment (michaelmattig) [reply operators/test-data/vector/data/points_with_iso_time.json:62]: this was just an inconsistency in the test data. they weren't actually all iso and now they are.

File: services/src/handlers/datasets.rs [hunk 6]
```diff
@@ -295,13 +291,8 @@ fn auto_detect_meta_data_definition(main_file_path: &Path) -> Result<MetaDataDef
...
-    let feature = layer.features().next();
-    if feature.is_none() {
-        return OgrSourceDatasetTimeType::None;
-    }
-    let feature = feature.expect("checked before");
-
+fn detect_time_type(columns: &Columns) -> OgrSourceDatasetTimeType {
+    // TODO: load candidate names from config
     let known_start = [
         "start",
```
[the remark this answers is not in the document]
Comment (michaelmattig) [reply services/src/handlers/datasets.rs:297]: ec80627aa13d252b5cfdd20dfa541a50a3a10ae4

File: services/src/handlers/datasets.rs [hunk 7]
```diff
@@ -316,6 +307,7 @@ fn detect_time_type(layer: &Layer, columns: &Columns) -> OgrSourceDatasetTimeTyp
         "date time",
         "event",
         "timestamp",
+        "time_from",
     ];
     let known_end = [
         "end",
```
[the remark this answers is not in the document]
Comment (michaelmattig) [reply services/src/handlers/datasets.rs:315]: ec80627aa13d252b5cfdd20dfa541a50a3a10ae4

**Pull request merged**